### PR TITLE
fix(gltf): fix of getTypedArrayForAccessor in gltf-scenegraph

### PR DESCRIPTION
**Pull request by mspivak-actionengine**
Regarding the module gltf-scenegraph.ts:
Functions getTypedArrayForAccessor and getTypedArrayForImageData are implemented incorrectly. These functions use method getBuffer that returns a buffer object from gltf.json, not from gltf. It doesn't have either buffer data or "data" property, which these functions are trying to get data from. They don't use binChunk.byteOffset either.
I implemented a separate function getTypedArrayForAccessor in module get-typed-array.ts.
This module already has the implementation of getTypedArrayForBufferView and getTypedArrayForImageData with the comment:
// TODO - GLTFScenegraph should use these
So, with this PR, getTypedArray** related functions in gltf-scenegraph.ts start using the corresponding functions from get-typed-array.ts.

### CODE REVIEW

**Review by ibgreen — COMMENTED** [review]
When making a change in the core parts a big code base like this, a reviewer like me worries about impact on existing code and how this new code path will be maintained.

A technique that helps at least with the latter is:

1) First, add new test cases covering the detected problem. They should fail on the existing code base (these can be added in a small separate PR).
2) Then put up the fix and show that is fixes those issues.

**Comment by mspivak-actionengine** [comment]
The test is in a separate PR: https://github.com/visgl/loaders.gl/pull/2695

File: modules/gltf/src/lib/api/gltf-scenegraph.ts [hunk 1]
```diff
@@ -19,11 +19,9 @@ import type {
...
 import {padToNBytes, copyToArray} from '@loaders.gl/loader-utils';
 import {assert} from '../utils/assert';
-import {
-  getAccessorArrayTypeAndLength,
-  getAccessorTypeFromSize,
-  getComponentTypeFromArray
-} from '../gltf-utils/gltf-utils';
+import {getAccessorTypeFromSize, getComponentTypeFromArray} from '../gltf-utils/gltf-utils';
+
+import {getTypedArrayForAccessor as _getTypedArrayForAccessor} from '../gltf-utils/get-typed-array';
```
Comment (ibgreen) [inline modules/gltf/src/lib/api/gltf-scenegraph.ts:24]: Why are we renaming it?

**Review by ibgreen — COMMENTED** [review]
Is this a combo of the previous PRs? Why not land them one by one? It is hard to review everything together like this.

**Review by ibgreen — APPROVED** [review]
This still got pretty big, in spite of all the prep PRs. I am approving but TBH, I still don't really feel I understand what is going on.

**Pull request merged**